source/client/output_formatter_impl.cc

```diff
@@ -138,5 +144,125 @@ DottedStringOutputFormatterImpl::formatProto(const nighthawk::client::Output& ou
   return ss.str();
 }
 
+const nighthawk::client::Result&
+FortioOutputFormatterImpl::getGlobalResult(const nighthawk::client::Output& output) const {
+  for (const auto& nh_result : output.results()) {
+    if (nh_result.name() == "global") {
+      return nh_result;
+    }
+  }
+
+  throw NighthawkException("Nighthawk output was malformed, contains no 'global' results.");
+}
+
+const nighthawk::client::Counter&
+FortioOutputFormatterImpl::getCounterByName(const nighthawk::client::Result& result,
+                                            absl::string_view counter_name) const {
+  for (const auto& nh_counter : result.counters()) {
+    if (nh_counter.name() == counter_name) {
+      return nh_counter;
+    }
+  }
+
+  throw NighthawkException(absl::StrCat(
+      "Nighthawk result was malformed, contains no counter with name: ", counter_name));
+}
+
+const nighthawk::client::Statistic& FortioOutputFormatterImpl::getRequestResponseStatistic(
+    const nighthawk::client::Result& result) const {
+  for (auto const& nh_stat : result.statistics()) {
+    if (nh_stat.id() == "benchmark_http_client.request_to_response") {
+      return nh_stat;
+    }
+  }
+
+  throw NighthawkException("Nighthawk result was malformed, contains no "
```

Comment:
This seems an anti-pattern, passing around expected output results as string map values.. should we have more structure here?

Reply:
@oschaaf perhaps you could look into this? I think this would require changes outside of my PR.

Reply:
I filed https://github.com/envoyproxy/nighthawk/issues/195 to discuss.

Reply:
(and track)